18-tistos-diy.md

```diff
@@ -0,0 +1,120 @@
+
+Let's use the TriggerTisTos tool now. In preparation for Run-II, the Hlt1 and
+Hlt2 DecReports and SelReports are now stored in different locations. That means
+two TisTos tools will be needed, each configured to pick up information from
+either HLT1 or HLT2. Since the tools we create are public tools, they have to be
+configured in the following way (before the AppMgr is instantiated):
+
+~~~ {.python}
+from Configurables import ToolSvc, TriggerTisTos
+ToolSvc().addTool(TriggerTisTos, "Hlt1TriggerTisTos")
+ToolSvc().Hlt1TriggerTisTos.HltDecReportsLocation = 'Hlt1/DecReports'
+ToolSvc().Hlt1TriggerTisTos.HltSelReportsLocation = 'Hlt1/SelReports'
+~~~
+
+Create the tools in the same way you created others during the
+[other lesson](05-interactive-dst.html), but use instance-specific names that
+correspond to the configuration we just added: "Hlt1TriggerTisTos" and
+"Hlt2TriggerTisTos". The tools use ITriggerTisTos as an interface.
+
+Use the advance function to find an event that has some candidates for the
+chosen selection and set the TisTos tools to use our candidate and trigger selection:
+
+~~~ {.python}
+hlt1TisTosTool.setOfflineInput()
+candidate = candidates[0]
+hlt1TisTosTool.addToOfflineInput(candidate)
+hlt1TisTosTool.setTriggerInput()
+hlt1TisTosTool.addToTriggerInput("Hlt1TrackAllL0Decision")
+result = hlt1TisTosTool.tisTosTobTrigger()
+result.tos()
+~~~
+
+The "set" calls reset the internal storage of candidate or trigger information,
+and the "addTo" calls then add the things we are interested in.
+
+An offline candidate is considered to be Tis with respect to a trigger selection
+if removing it from the event would still cause the trigger selection to accept
+the event, i.e. if there is another particle in the event that was also
+accepted by the trigger selection. In more formal terms, if the LHCbIDs of the
+all of the final state particles of any of the candidates accepted by the
+trigger selection overlap less than 1% with all of the LHCbIDs of the final
+state particles of the offline candidate.
+
+~~~ {.python}
+result.tis()
+~~~
+
+Note that a candidate can be both Tis and Tos with respect to a trigger
+selection, or Tos with respect to one selection, and Tis with respect to
+another. To tell the tool to consider more trigger selections, use the following
+(regexes are also supported), and try to find some events that are both Tos and
+Tis:
+
+~~~ {.python}
+hlt1TisTosTool.setTriggerInput()
+hlt1TisTosTool.addToTriggerInput("Hlt1TrackAllL0Decision")
+hlt1TisTosTool.addToTriggerInput("Hlt1DiMuonHighMassDecision")
+result = hlt1TisTosTool.tisTosTobTrigger()
+~~~
+
+The (Tos) trigger efficiency of a trigger selection can be calculated as:
+
+eps\_{Tos} = N\_{Tis AND Tos} / N\_{Tis}
+
+Loop over the events in the DST and calculate the efficiency of
+Hlt1TrackAllL0. You can add some more Hlt1 selecitons when checking for Tis,
+which ones would make sense?
```

Comment:
(Why do you not use "all other physics" triggers when looking for Tis? This is something I never understood. Maybe I should ask during the workshop ;) )